jgiven-maven-plugin/src/test/java/com/tngtech/jgiven/maven/BasicMavenMojoTest.java

```diff
@@ -1,85 +1,119 @@
 package com.tngtech.jgiven.maven;
 
+import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.shared.invoker.*;
-import org.gradle.testkit.runner.GradleRunner;
-import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.HashMap;
 import java.util.List;
 import java.util.stream.Stream;
 
+import static java.lang.String.format;
 import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
 import static org.assertj.core.api.Assertions.assertThat;
 
 class BasicMavenMojoTest {
 
     private static final String PROJECT_FOLDER = "sampleProject"; //name is also used in renovate.json
-    @TempDir
-    public Path temporaryDirectory;
 
-    @BeforeEach
-    void copyResourcesToTemporaryDirectory() throws IOException {
+    private static File projectFile;
+
+    @BeforeAll
+    static void prepareTestProject(@TempDir File temporaryDirectory) throws Exception {
+        projectFile = new File(temporaryDirectory, "project");
+
         var sourceFolder = Path.of("src/test/resources", PROJECT_FOLDER);
-        copyFolder(sourceFolder, temporaryDirectory);
+        copyFolder(sourceFolder, projectFile.toPath());
+
+        buildMavenTestProject();
     }
 
-    @BeforeEach
-    void publishPluginVersionToMavenLocal() {
-        var env = new HashMap<>(System.getenv());
-        env.put("RELEASE", "false");
-        GradleRunner.create().withProjectDir(new File(System.getProperty("user.dir")))
-                .withEnvironment(env)
-                .withArguments(
-                        "publishToMavenLocal",
-                        "-x", "test" //don't test, or we'll loop infinitely
-                )
-                .build();
+    static Stream<Arguments> provideFormatsAndExpectedOutput() {
+        return Stream.of(Arguments.of("html", "index.html"),
+                Arguments.of("asciidoc", "index.asciidoc"),
+                Arguments.of("text", "testpackage.ThingDoerTest.feature")
+        );
     }
 
-    @Test
-    void testMavenProducesHtmlAndAsciiReport() throws MavenInvocationException {
-        var mavenExecutable = this.findMavenExecutable();
+    @ParameterizedTest
+    @MethodSource("provideFormatsAndExpectedOutput")
+    void should_determine_output_dir_based_on_format(String format, String expectedOutput) throws MojoExecutionException {
+        var mavenMojo = new JGivenReportMojo(
+                new File(projectFile, "target"),
+                null,
```

Comment:
I suppose we can't avoid a null here?

Reply:
In fact, that particular `null` is even necessary, because we're testing that the MOJO chooses the correct output directory